libraries/NonFungibleToken/contracts/non-fungible.aes

```diff
@@ -1,110 +1,75 @@
-    ensureAllowed(Call.caller)
-    put(state{operatorApprovals[Call.caller][to] = approved})
-    true
-  
-  private function checkTokenExists(tokenId : int) : bool =
-    switch(Map.lookup(tokenId, state.tokenOwner))
-      None    => false
-      Some(x) => true
-
-  private function ensureAllowed(key : address) =
-    switch(Map.lookup(key, state.operatorApprovals))
-      None => put(state{operatorApprovals[key] = {}})
-      Some(_) => ()
-
-  private stateful function addTokenTo(to : address, tokenId : int) = 
-    put(state{tokenOwner = state.tokenOwner{[tokenId] = to}})
-    put(state{ownedTokensCount[to] = add(lookupByAddress(to, state.ownedTokensCount, 0), 1)})
-
-  private stateful function removeTokenFrom(from : address, tokenId : int) =
-    put(state{tokenOwner = state.tokenOwner{[tokenId] = #0}})
-    put(state{ownedTokensCount[from] = sub(lookupByAddress(from, state.ownedTokensCount, 0), 1)})
-
-  private stateful function clearApproval(from : address, tokenId : int) : bool = 
-    put(state{tokenApprovals = state.tokenApprovals{[tokenId] = #0}})
-    true
-
-  private function lookupByAddress(k : address, m, v) =
-    switch(Map.lookup(k, m))
-      None    => v
-      Some(x) => x
-
-  private function lookupById(k : int, m, v) =
-    switch(Map.lookup(k, m))
-      None    => v
-      Some(x) => x
-
-  private function add(_a : int, _b : int) : int =
-    let c : int = _a + _b
-    require(c >= _a, "Error")
-    c
-
-  private function sub(_a : int, _b : int) : int =
-    require(_b =< _a, "Error")
-    _a - _b
-
-  //todo use native abort!!
-  private function abort(err) = abort(err)
-
-  private function require(b : bool, err : string) =
-    if(!b) 
-      abort(err)
-
-  private function onlyOwner(owner : address) =
-    if(owner != state.owner) 
-      abort("Only owner can mint!")
-
-  private function onlyTokenOwner(owner : address, tokenId : int) =
-    if(owner != lookupById(tokenId, state.tokenOwner, #0)) 
-      abort("Only token owner can transfer!")
-
-  private function isApprovedOrOwner(spender : address, tokenId : int) : bool = 
-    let owner : address = ownerOf(tokenId)
-    getApproved(tokenId) == spender || owner == spender || isApprovedForAll(owner, spender)
\ No newline at end of file
+    clear_approval(from, token_id)
+    remove_token_from(from, token_id)
+    add_token_to(to, token_id)
+
+  public stateful function set_approval_for_all(to: address, approved: bool) =
+    put(state{m_operator_approvals[Call.caller = {}][to] = approved})
+
+  private stateful function add_token_to(to: address, token_id: int) = 
+    put(state{m_tokenOwner = state.m_tokenOwner{[token_id] = to}})
+    put(state{m_owned_tokens_count[to] = SafeMath.add(Map.lookup_default(to, state.m_owned_tokens_count, 0), 1)})
+
+  private stateful function remove_token_from(from: address, token_id: int) =
+    put(state{m_tokenOwner = state.m_tokenOwner{[token_id] = #0}})
```

Comment:
Change

`put(state{m_tokenOwner = state.m_tokenOwner{[token_id] = #0}})`

to 

`put(state{m_token_owner = state.m_token_owner{[token_id] = #0}})`

Reply:
what is the `m_` semantic?

Reply:
I was asking the same here:
https://github.com/aeternity/aepp-sophia-examples/pull/37#discussion_r276681580

My comment here is to point out that snake_case should be used consistently. `m_tokenOwner` => `m_token_owner`

Reply:
the **m_** prefix is used to differentiate Maps from other variables

Reply:
> the **m_** prefix is used to differentiate Maps from other variables

Is it necessary?

Reply:
In my opinion, yes it is necessary and it is easier to read if you can make this difference

Reply:
but why m_ instead if map_ or another semantic to let devs know

Reply:
is this more readable: 
```
private function foldr(f : (('a, 'b) => 'b), z: 'b, l : list('a)) : 'b =
    switch(l)
      [] => z
      e :: l' => f(e, foldr(f, z, l'))
```
what is `z`? 
or should we type full name always like f => `function'` and l => `list`

Reply:
@thepiwo @mradkov we thought the **m_** prefix is descriptive enough, and if we use **map** will make the variables too long.

Reply:
> @thepiwo @mradkov we thought the **m_** prefix is descriptive enough, and if we use **map** will make the variables too long.

I thought it was used for this after looking over the code in the first place, but in Solidity, for example, nobody is using something similar when naming variables and properties.

I think using something like this, here in the Sophia examples, we are pretty much influencing the language convention that should be used by developers when developing on aeternity.

That being said - in my opinion using the longer `map_` prefix shouldn't be a problem - it doesn't affect price, fees or something else. So if we proceed with the prefix at all - I'm vouching for `map_`.

 After all the code should be easily readable for humans. :)

Reply:
`z` is the starting value, the initial value of the accumulator

Reply:
`list` and `function` are reserved keywords, I guess they could be used, but for library functions that represent no business logic I did choose to keep them as short as possible